apigw-custom-domain/README.md

```diff
@@ -1,70 +1,69 @@
+| Categories   | Serverless; REST API              |
+| Level        | Beginner                          |
+
+## Introduction
+
+A demo application showcasing API Gateway (v2) endpoints with custom domain names configured through Route53 and ACM, deployed locally using LocalStack and the Serverless framework. For more details, refer to the [documentation](https://docs.localstack.cloud/user-guide/aws/apigateway/#custom-domain-names-with-api-gateway).
 
 ## Prerequisites
 
-* LocalStack
+* [`localstack` CLI](https://docs.localstack.cloud/getting-started/installation/#localstack-cli) with [`LOCALSTACK_AUTH_TOKEN`](https://docs.localstack.cloud/getting-started/auth-token/)
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) with [`awslocal` wrapper script](https://docs.localstack.cloud/user-guide/integrations/aws-cli/#localstack-aws-cli-awslocal)
+* [Node.js 18.x](https://nodejs.org/en/download/package-manager) with `npm`
+* [Serverless Framework](https://www.serverless.com/framework/docs/getting-started) 3.x
 * Docker
-* Node.js / `npm`
-* `make`
+* `openssl` & `make`
 
-## Installing
+You can run `make check` to verify that all dependencies are installed.
 
-To install the dependencies:
-```
+## Installation
+
+Run the following command to install the necessary dependencies:
+
+```bash
 make install
 ```
 
-## Running
+## Start LocalStack
 
-Make sure that LocalStack is started:
-```
+To start LocalStack, run the following command:
+
+```bash
 LOCALSTACK_AUTH_TOKEN=... DEBUG=1 localstack start
 ```
 
-Deploy the app locally and run a test invocation via:
-```
-make run
-```
+## Deploy the Application
 
-The script first generates an SSL certificate for local testing (in case the `openssl` command is not available, it will use an existing, predefined certificate), and then adds it to Amazon Certificate Manager (ACM), and finally creates a Route53 hosted zone for the domain name `test.example.com`:
-```
-Generating a 2048 bit RSA private key
-...
-subject=/CN=test.example.com
-...
-Importing local test certificate into ACM API ...
-{
-    "CertificateArn": "arn:aws:acm:us-east-1:000000000000:certificate/9cbc69d6-abf9-412e-9e2b-36f99fcbf251"
-}
-Creating Route53 hosted zone for test domain 'test.example.com' ...
-{
-    "HostedZone": {
-        "Id": "/hostedzone/SU1TPRNX6CL3OE0",
-        "Name": "test.example.com.",
-        ...
-```
+Deploy the application locally using the Serverless framework:
 
-Next, you should see some output with the deployment logs of the Serverless application, and some details in the output section towards the bottom:
 ```
-...
-Serverless Domain Manager: Info: Created API mapping '(none)' for test.example.com
-Serverless Domain Manager: Summary: Distribution Domain Name
-Serverless Domain Manager:    Domain Name: test.example.com
-Serverless Domain Manager:    Target Domain: test.example.com
-Serverless Domain Manager:    Hosted Zone Id: Z2FDTNDATAQYW2
+./run.sh
```

Comment:
is there a reason to invoke this script directly, instead of using the make run command?

Reply:
`make run` invokes the smoke test as well with it.

Reply:
I envision that for local testing, we should rely more on shell scripts and use make targets for CI testing

Reply:
I'm in agreement with @bartsdev and @joe4dev here.